vendor/src/github.com/docker/engine-api/types/container/config.go

```diff
@@ -5,6 +5,20 @@ import (
 	"github.com/docker/go-connections/nat"
 )
 
+// HealthConfig holds configuration settings for the HEALTHCHECK feature.
+// For durations, a value of -1 means to inherit the default value.
+type HealthConfig struct {
+	Test        strslice.StrSlice // Test to perform to check container is healthy.
+				      // An empty slice means to inherit the default.
+				      // Otherwise, Cmd[0] is "NONE" or "CMD".
+	GracePeriod float64           // Time after container start to assume failure means still starting.
```

Comment:
Unfortunately, Go converts durations to JSON by printing its internal format (integer nanoseconds), which would leak an implementation detail of the Go language into the protocol, so I converted to floating-point seconds instead as a more standard unit.